### PR TITLE
Add multi resource rule example

### PR DESCRIPTION
**Pull request by daniel-butler-irl**
### Description

Adds an example to show applying a rule to multiple resources

### Release required?

- [x] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- When merging, use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

File: modules/cbr-mutli-rule-module/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Multi CBR Rule Module
+
+Creates multiple rules for Context Based Restrictions
+
```
Comment (ocofaigh) [inline modules/cbr-mutli-rule-module/README.md:4]: Can you add usage here?

Comment (daniel-butler-irl) [reply modules/cbr-mutli-rule-module/README.md:4]: Sure I'll add it. As an FYI the other submodules do not have usage

File: examples/multi-rule/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Multi-rule example
+
+An end-to-end example that uses the module's default variable values. This example uses the IBM Cloud Provider to automate the following infrastructure::
```
Comment (ocofaigh) [inline examples/multi-rule/README.md:3]: remove one of the colons `:`

File: modules/cbr-mutli-rule-module/README.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+## Usage
+
+```hcl
+module "cbr_rules" {
+  source           = "terraform-ibm-modules/cbr/ibm//modules/cbr-rule-module"
```
Comment (ocofaigh) [inline modules/cbr-mutli-rule-module/README.md:9]: this is wrong source

File: modules/cbr-mutli-rule-module/README.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+  version          = "1.9.0"
```
Comment (ocofaigh) [inline modules/cbr-mutli-rule-module/README.md:10]: should we update to:
`version = "X.X.X" # Replace "X.X.X" with a release version to lock into a specific release`

File: modules/cbr-mutli-rule-module/main.tf [hunk 5]
```diff
@@ -0,0 +1,15 @@
+##############################################################################
+# Context Based Restrictions module
+#
+# Creates CBR Rules
+##############################################################################
+module "cbr_rules" {
+  count            = length(var.rule_list)
+  source           = "terraform-ibm-modules/cbr/ibm//modules/cbr-rule-module"
+  version          = "1.9.0"
```
Comment (ocofaigh) [inline modules/cbr-mutli-rule-module/main.tf:9]: This should be using a local source as the module exists in same repo

File: examples/multi-rule/main.tf [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+      kms_encryption_enabled = false
+      region_location        = var.region
+      resource_group_id      = module.resource_group.resource_group_id
+      resource_instance_id   = module.cos_instance.cos_instance_id
+    }
+  ]
+}
+
+locals {
+  # Filter out empty tags
```
Comment (ocofaigh) [inline examples/multi-rule/main.tf:141]: I'm a little lost trying to figure out whats going on with all this bucket tag logic. Can you maybe add some more code comments which explain why and what the code is doing?

**Review by ocofaigh — CHANGES_REQUESTED** [review]
See comments

**Review by ocofaigh — CHANGES_REQUESTED** [review]
Can we add a test that call this example to ensure it remains a working example? Doesn't have to be in pr_test.go - can go into other_test.go and run weekly

File: README.md [hunk 7]
```diff
@@ -62,6 +63,16 @@ You need the following permissions to run this module.
...
+<!-- END MODULE HOOK -->
+<!-- BEGIN EXAMPLES HOOK -->
+## Examples
+
+- [ Pre-wired CBR configuration for FS Cloud example](examples/fscloud)
+- [ Multi resource rule example](examples/multi-resource-rule)
+- [ CBR multi service profile](examples/multi-service-profile)
+- [ Multi-zone example](examples/multizone-rule)
+- [ Zone example](examples/zone)
+<!-- END EXAMPLES HOOK -->
```
Comment (ocofaigh) [inline README.md:75]: This hook is depreciated as examples are part of Overview hook. Remove it from here

**Review by ocofaigh — APPROVED** [review]

**Review by ocofaigh — CHANGES_REQUESTED** [review]
sorry, didn't mean to approve - one comment to address https://github.com/terraform-ibm-modules/terraform-ibm-cbr/pull/307#discussion_r1352592138

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**